stdlib/asm/crypto/dsa/rpo_falcon512.masm

```diff
@@ -218,182 +207,83 @@ export.load_h_s2_and_product.4
         push.M u32lt assert
```

Comment:
Not from this PR, but could this be done as:
```
u32overflowing_sub.M assert drop
```
If this works, it should save 1 cycle per comparison or 512 cycles total.

Also, add add a comment above line 194 above saying that here we are checking that all coefficients are smaller than Falcon prime.

Reply:
Good point, we are indeed just range checking the coefficients. Changed to the more optimized suggestion.

Added the comment above line 194